adminSiteClient/DatasetEditPage.tsx

```diff
@@ -148,26 +143,6 @@ class DatasetEditor extends Component<{ dataset: DatasetPageData }> {
         }
     }
 
-    async delete() {
```

Comment:
I think this could still be useful, e.g. after we migrate a dataset to ETL (by creating a new one with the same name) we want to delete the old non-ETL one.

Reply:
Okay! It's just that there isn't any button in the UI for this (the only button is "Archive"). Should we add that button, then? Or remove the front-end logic, but not the back-end route for it?

Reply:
Oooh, right :D Then delete it, archiving is a better option in all cases anyway.